dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -272,12 +274,123 @@ Int64 WindowTransformAction::getPartitionEndRow(size_t block_rows)
     return left;
 }
 
+RowNumber WindowTransformAction::stepForward(const RowNumber & current_row, UInt64 n)
+{
+    auto dist = distance(current_row, partition_start);
+    assert(dist >= 0);
```

Comment:
Because this depends the arg of Frame?
```suggestion
    RUNTIME_CHECK(dist >= 0);
```